src/snowflake/connector/connection_diagnostic.py

```diff
@@ -214,6 +214,12 @@ def __test_socket_get_cert(
                 context = ssl.SSLContext(ssl.PROTOCOL_SSLv23)
                 sock = context.wrap_socket(conn, server_hostname=host)
                 certificate = ssl.DER_cert_to_PEM_cert(sock.getpeercert(True))
+                http_request = f"""GET / {host}:{port} HTTP/1.1\r\n
+                                   Host: {host}\r\n
+                                   User-Agent: snowflake-connector-python-diagnostic
+                                   \r\n\r\n"""
+                sock.send(str.encode(http_request))
+                sock.recv(4096).decode("utf-8")
```

Comment:
I'm not familiar with the diagnostic workflow, is the `sock.recv(4096).decode("utf-8")` necessary here?
also do you want try-except here in case `sock.send` or `sock.recv` failed but you still the flow keep moving forward even without http agent successfully, or errors already handled in upper stream?

Reply:
I swore when I wrote this the sock recv was needed, but it does not seem to be the case.  I've removed that.  I do have the try/except in the methods that call this one, but added the extra layer just in case.